src/SignupForm.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+
+class SignupForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      firstName: '',
+      lastName: '',
+      password: '',
+      agreeWithTerms: false,
+    };
+
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
+    this.handleLastNameChange = this.handleLastNameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.toogleAgreeWithTerms = this.toogleAgreeWithTerms.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleEmailChange(e) {
+    this.setState({ email: e.target.value })
```

Comment:
Can you merge these input changing functions into one? There's no need for 4 different functions

Reply:
It should be possible.